lib/pbench/test/unit/server/test_shell_cli.py

```diff
@@ -0,0 +1,384 @@
+        monkeypatch.setattr(shell, "init_db", init_db)
+
+        ret_val = shell.main()
+
+        assert ret_val == 1
+
+    @staticmethod
+    def test_main_wait_for_oidc_server_exc(
+        monkeypatch, make_logger, mock_get_server_config
+    ):
+        def wait_for_oidc_server(
+            server_config: PbenchServerConfig, logger: logging.Logger
+        ) -> str:
+            raise Exception("oidc exception")
+
+        monkeypatch.setattr(shell.site, "ENABLE_USER_SITE", False)
+        monkeypatch.setattr(
+            shell.OpenIDClient, "wait_for_oidc_server", wait_for_oidc_server
+        )
+
+        ret_val = shell.main()
+
+        assert ret_val == 1
+
+    @staticmethod
+    def test_main_wait_for_database_exc(
+        monkeypatch, make_logger, mock_get_server_config
+    ):
+        def wait_for_database(
+            server_config: PbenchServerConfig, logger: logging.Logger
+        ) -> str:
+            raise ConnectionRefusedError("database exception")
+
+        monkeypatch.setattr(shell.site, "ENABLE_USER_SITE", False)
+        monkeypatch.setattr(shell.Database, "wait_for_database", wait_for_database)
+
+        ret_val = shell.main()
+
+        assert ret_val == 1
+
+    @staticmethod
+    @pytest.mark.parametrize("section", ["database", "pbench-server"])
+    def test_main_server_config_no_section(
+        monkeypatch, on_disk_server_config, make_logger, section
+    ):
+        def get_server_config() -> PbenchServerConfig:
+            cfg_file = on_disk_server_config["cfg_dir"] / "pbench-server.cfg"
+            config = PbenchServerConfig(str(cfg_file))
+            del config.conf[section]
+            return config
+
+        monkeypatch.setattr(shell, "get_server_config", get_server_config)
+
+        ret_val = shell.main()
+
+        assert ret_val == 1
+
+    @staticmethod
+    @pytest.mark.parametrize(
+        "option",
+        [
+            "bind_host",
+            "bind_port",
+            "uri",
+            "wait_timeout",
+            "workers",
+            "worker_timeout",
+            "crontab-dir",
+        ],
+    )
+    def test_main_server_config_no_option(
+        monkeypatch, on_disk_server_config, make_logger, option
+    ):
+        def get_server_config() -> PbenchServerConfig:
+            cfg_file = on_disk_server_config["cfg_dir"] / "pbench-server.cfg"
+            config = PbenchServerConfig(str(cfg_file))
+            section = (
+                "database"
+                if option in frozenset(("uri", "wait_timeout"))
+                else "DEFAULT"
+                if option == "crontab-dir"
+                else "pbench-server"
+            )
+            make_logger.error(config.conf[section][option])
+            del config.conf[section][option]
+            return config
+
+        monkeypatch.setattr(shell, "get_server_config", get_server_config)
+
+        ret_val = shell.main()
+
+        assert ret_val == 1
```

Comment:
Given that this unit test doesn't check the log, should the mock at line 356 be logging that "error"?

Reply:
I found it useful debugging.